DMX.Portal.Web.Tests.Unit/Services/Foundations/Labs/LabServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,140 @@
+
+            dmxApiBrokerMock.Verify(broker =>
+                broker.PostLabAsync(It.IsAny<Lab>()),
+                    Times.Never);
+
+            dmxApiBrokerMock.VerifyNoOtherCalls();
+            loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task ShouldThrowValidationExceptionOnAddIfLabIsInvalidAndLogItAsync(
+            string invalidString)
+        {
+            // given
+            var invalidLab = new Lab()
+            {
+                Name = invalidString,
+                Description = invalidString,
+            };
+
+            var invalidLabException = new InvalidLabException();
+
+            invalidLabException.AddData(
+                key: nameof(Lab.Id),
+                values: "Id is required");
+
+            invalidLabException.AddData(
+                key: nameof(Lab.Name),
+                values: "Text is required");
+
+            invalidLabException.AddData(
+                key: nameof(Lab.Description),
+                values: "Text is required");
+
+            var expectedLabValidationException = 
+                new LabValidationException(invalidLabException);
+
+            // when
+            ValueTask<Lab> addLabTask = 
+                this.labService.AddLabAsync(invalidLab);
+
+            LabValidationException actualLabValidationException =
+                await Assert.ThrowsAsync<LabValidationException>(addLabTask.AsTask);
+
+            // then
+            actualLabValidationException.Should().BeEquivalentTo(expectedLabValidationException);
+
+            loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedLabValidationException))),
+                Times.Once);
+
+            dmxApiBrokerMock.Verify(broker =>
+                broker.PostLabAsync(It.IsAny<Lab>()),
+                Times.Never);
+
+            dmxApiBrokerMock.VerifyNoOtherCalls();
+            loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfLabStatusIsInvalidAndLogItAsync()
+        {
+            // given
+            var randomLab = CreateRandomLab();
+            Lab invalidLab = randomLab;
+            invalidLab.Status = GetInvalidEnum<LabStatus>();
+
+            var invalidLabException = new InvalidLabException();
+
+            invalidLabException.AddData(
+                key: nameof(Lab.Status),
+                values: "Value is not recognized");
+
+            var expectedLabValidationException =
+                new LabValidationException(invalidLabException);
+
+            // when
+            ValueTask<Lab> addLabTask =
+                this.labService.AddLabAsync(invalidLab);
+
+            LabValidationException actualLabValidationException =
+                await Assert.ThrowsAsync<LabValidationException>(addLabTask.AsTask);
```

Comment:
This validation doesn't belong here at all.
Let me clarify:
Your Gatekeeper and Portal do a different set of primitive validations than your Core Api.
So your portal here only cares about structural validations which you've already done, except for this enum. Portal doesn't care about the enum value being logical as in range or not.